yt_dlp/extractor/crunchyroll.py

```diff
@@ -155,28 +155,24 @@ class CrunchyrollBetaIE(CrunchyrollBaseIE):
 
     def _real_extract(self, url):
         lang, internal_id, display_id = self._match_valid_url(url).group('lang', 'id', 'display_id')
-        api_domain, bucket, params = self._get_params(lang)
 
-        episode_response = self._download_json(
-            f'{api_domain}/cms/v2{bucket}/episodes/{internal_id}', display_id,
-            note='Retrieving episode metadata', query=params)
-        if episode_response.get('is_premium_only') and not episode_response.get('playback'):
+        episode_response = traverse_obj(self._call_api(
+            f'episodes/{internal_id}', display_id, lang, 'episode'), ('data', 0))
+        if episode_response.get('is_premium_only') and not episode_response.get('streams_link'):
             if self.is_logged_in:
                 raise ExtractorError('This video is for premium members only', expected=True)
             else:
                 self.raise_login_required('This video is for premium members only')
 
-        stream_response = self._download_json(
-            f'{api_domain}{episode_response["__links__"]["streams"]["href"]}', display_id,
-            note='Retrieving stream info', query=params)
-        get_streams = lambda name: (traverse_obj(stream_response, name) or {}).items()
-
         requested_hardsubs = [('' if val == 'none' else val) for val in (self._configuration_arg('hardsub') or ['none'])]
         hardsub_preference = qualities(requested_hardsubs[::-1])
         requested_formats = self._configuration_arg('format') or ['adaptive_hls']
```

Comment:
It seems a bit confusing to move this before the stream api call, but still have the episode api request above it. Was there a particular reason you reordered it?

We could put it at the top, or just before it's actually needed in the steam response processing, but having it in the middle seems odd.